docs/how-to-guides/convert-changelog-to-table.md

```diff
@@ -10,113 +10,139 @@ keywords: changelog, table, materialize
+```sql
+SELECT * FROM latest WHERE k = 'k2';
++------+------+------+------+
+|K     |V1    |V2    |V3    |
++------+------+------+------+
+|k2    |4     |e     |true  |
+```
+
+
+## Materializing a changelog `STREAM`
+
+The best way to materialize an input topic as a table is the `CREATE SOURCE TABLE` statement. If you already have a `STREAM`, though, you can also materialize it as a table with the `CREATE TABLE ... AS ...` statement. Let's say you have a `STREAM` of events in ksqlDB that represent a series of changes called `changelog_stream`, and you want a view of the data that reflects the latest values for each key. 
+
+Begin by telling ksqlDB to start all queries from the earliest point in each topic:
+```sql
+SET 'auto.offset.reset' = 'earliest';
+```
 
+Let's mimic adding a few records to the `changelog_stream` by using the `INSERT INTO` statement:
 ```sql
-INSERT INTO s1 (
+INSERT INTO changelog_stream (
     k, v1, v2, v3
 ) VALUES (
     'k1', 0, 'a', true
 );
 
-INSERT INTO s1 (
+INSERT INTO changelog_stream (
     k, v1, v2, v3
 ) VALUES (
     'k2', 1, 'b', false
 );
 
-INSERT INTO s1 (
+INSERT INTO changelog_stream (
     k, v1, v2, v3
 ) VALUES (
     'k1', 2, 'c', false
 );
 
-INSERT INTO s1 (
+INSERT INTO changelog_stream (
     k, v1, v2, v3
 ) VALUES (
     'k3', 3, 'd', true
 );
 
-INSERT INTO s1 (
+INSERT INTO changelog_stream (
     k, v1, v2, v3
 ) VALUES (
     'k2', 4, 'e', true
 );
 ```
 
-Derive a table, `t1`, from stream `s1`. When you create a table, the columns in the `SELECT` clause must either be the column that you're grouping by or columns with an aggregation function applied. The `LATEST_BY_OFFSET` aggregation allows you to select any column and retains only the last value it receives, where "last" is in terms of offsets.
+You can view the data in the `changelog_stream` by issuing a pull query:
+```sql
+SELECT * FROM changelog_stream;
++------+------+------+------+
+|K     |V1    |V2    |V3    |
++------+------+------+------+
+|k1    |0     |a     |true  |
+|k2    |1     |b     |false |
+|k1    |2     |c     |false |
+|k3    |3     |d     |true  |
+|k2    |4     |e     |true  |
+Query terminated
+```
 
+To view the data that reflects only the last change for each key in `changelog_stream`, you can derive a table called `latest_events` with a `CREATE TABLE AS SELECT` statement in conjunction with the `LATEST_BY_OFFSET` aggregation:
 ```sql
-CREATE TABLE t1 AS
+CREATE TABLE latest_events AS
     SELECT k,
            LATEST_BY_OFFSET(v1) AS v1,
            LATEST_BY_OFFSET(v2) AS v2,
            LATEST_BY_OFFSET(v3) AS v3
-    FROM s1
+    FROM changelog_stream
     GROUP BY k
     EMIT CHANGES;
 ```
 
-Run a pull query against each key. Notice how for each key, the columns reflect the last set of values that were inserted.
-
+Now, you can view the latest values for each key in your `changelog_stream` by issuing a pull query against the `latest_events` table that we just created above:
```

Comment:
Maybe add same information that the query will need some time to reach the end of the input stream and that pull queries might return "old" data if the end of the input stream is not reached yet?